src/OpenRiaServices.Client/Test/Client.Test/Data/OperationTests.cs

```diff
@@ -439,6 +442,91 @@ public void Bug706066_CancelInCallback()
             EnqueueTestComplete();
         }
 
+
+
+        class ExceptionHandleSynchronizationContext : SynchronizationContext
+        {
+            public Exception LastException { get; set; }
+            public override void Send(SendOrPostCallback d, object state)
+            {
+                if (state is ExceptionDispatchInfo exceptionDispatchInfo)
+                {
+                    LastException = exceptionDispatchInfo.SourceException;
+                }
+
+                base.Send(d, state);
+            }
+        }
+
+
+
+
+        /// <summary>
+        /// When operation complated callback occurr exception,
+        /// check if an exception was thrown on the SynchronizationContext.
+        /// </summary>
+        [TestMethod]
+        [Asynchronous]
+        public void Operation_CompleteCallbackHasError_ThrowSycCtx()
```

Comment:
You may not need to add a new testcasse if you update the 3 existingt test for when the callback throws exception.

Reply:
Thank you for view comment
I modified the 3 methods you suggested.


While referring to other methods for writing unit tests, I saw the following method.
The code below tests the callback, so how about modify this part?